editor/src/components/canvas/ui-jsx-canvas-renderer/canvas-render-errors.tsx

```diff
@@ -0,0 +1,57 @@
+import { FancyError } from '../../../core/shared/code-exec-utils'
+import { JSXElement } from '../../../core/shared/element-template'
+import { HighlightBoundsForUids } from '../../../core/shared/project-file-types'
+import { getUtopiaIDFromJSXElement } from '../../../core/shared/uid-utils'
+import StackFrame, { ScriptLine } from '../../../third-party/react-error-overlay/utils/stack-frame'
+
+export function canvasMissingJSXElementError(
+  factoryFunctionName: string | null,
+  sourceCode: string,
+  jsxElement: JSXElement,
+  filePath: string,
+  highlightBounds: HighlightBoundsForUids | null,
+): FancyError {
+  const error = new Error(
+    `(${jsxElement.name.baseVariable}) Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined.`,
+  )
+
+  const uid = getUtopiaIDFromJSXElement(jsxElement)
+  const highlightBoundsForElement = highlightBounds != null ? highlightBounds[uid] : null
+  const lineNumber = (highlightBoundsForElement?.startLine ?? 1) + 1
```

Comment:
that 1+1 looks suspicious to me :) 

```suggestion
  const lineNumber = (highlightBoundsForElement?.startLine ?? 0) + 1
```